File: pset1/more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_row_spec();
void mario_pyramid(int h);
void print_half_row(int n, int half_row, bool isSecond);
void print_spaces(int n);
void print_hashes(int n);

int main(void)
{
    int h = get_row_spec();
    if (h > 0)
    {
        mario_pyramid(h);
    }
}

int get_row_spec()
{
    printf("Input a height for the half-pyramid\n");
    int h = get_int();
    if (h <= 23 && h > 0)
    {
        return h;
    }
    else
    {
        printf("Please input an integer between 1 and 23\n");
        return 0;
    }
}

void mario_pyramid(int h)
{
   for (int i = 1; i <= h; i++)
    {
        print_half_row(i, h, false);
        print_spaces(2);
        print_half_row(i, h, true);
        printf("\n");
    }
}

void print_half_row(int n, int half_row, bool isSecond)
{
    int spaces = half_row - n;
    if (isSecond)
    {
        print_hashes(n);
        print_spaces(spaces);
    }
    else
    {
        print_spaces(spaces);
        print_hashes(n);
    }
}

void print_spaces(int n)
{
   for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

void print_hashes(int n)
{
   for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}

File: pset6/mario.py
import cs50

def printLoop(n, toPrint):
    for i in range(n):
        print(toPrint, end="")

def printHashes(n):
    printLoop(n, "#")

def printSpaces(n):
    printLoop(n, " ")

def printHalfRow(n, half_row, isSecond):
    spaces = half_row - n

    if isSecond:
        printHashes(n)
        printSpaces(spaces)

    else:
        printSpaces(spaces)
        printHashes(n)

def makeMarioPyramid(height):
    for i in range(1, height):
        printHalfRow(i, height, False)
        printSpaces(2)
        printHalfRow(i, height, True)
        print("")

def getRowSpec():
    print("Input a height for the half pyramid: ", end="")
    height = cs50.get_int()
    if height <= 23 and height > 0:
        return height
    else:
        print("Please input an integer between 1 and 23 inclusive")
        exit(1)

def main():
    height = getRowSpec()
    makeMarioPyramid(height)

if __name__ == "__main__":
    main()






